### DISCUSSION TITLE
distDir inside target does not get honored

### DISCUSSION THREAD

**Discussion by VincentVToscano** [parcel-bundler/parcel#6778] [Q&A]
Hi Devs & Community,

I am getting started with Parcel `2.0.0-rc.0` with a series of tests to verify if the workflow will be inline with my team's needs and ran into something that I was looking to get your help on. Enjoying the Parcel experience so far and am excited to get into more features soon. Thanks in advance.

:question:
I am experimenting with [*targets*,](https://v2.parceljs.org/features/targets/#targets) and I'd like to take advantage of the `distDir` option. I have been able to use this successfully when it is not in the *targets* object, but when I place inside the *targets* object, the default `dist` directory is created instead. I am running seeing this when I run the `start` command inside the package.json (the dev build).

## 🎛 Configuration
**package.json**
```
{
  "name": "ParcelTest001_2021",
  "version": "1.0.0",
  "description": "Parcel test",
  "repository": "unknown",
  "author": "Vincent",
  "license": "UNLICENSED",
  "private": true,
  "targets": {
    "frontend": {
      "source": "index.html",
      "distDir": "../build"
    }
  },
  "scripts": {
    "start": "parcel",
    "build": "parcel build"
  },
  "devDependencies": {
    "parcel": "^2.0.0-rc.0"
  }
}
```

## 📁 Directory & File Structure
```
*ParcelTest001_2021*
└── build
└── src
    ├── home.js
    ├── index.html
    ├── package.json
    └── yarn.lock
```

## 🤔 Expected Behavior
Bundled files should be outputted into the `build` directory

## 😯 Current Behavior
Bundled files are outputted in the `dist` directory

## 🔦 Context
The option `distDir` under frontend target is ignored.

## 🌍 Your Environment
Parcel: 2.0.0-rc.0
Node: v16.8.0
Yarn: yarn-3.0.1.cjs
Operating System : macOS current

Thanks for your help

**Comment by aakoch** [comment]
Does it behave the same when package.json is in the root directory (instead of src)? I'm not used to seeing the package.json file anywhere but the root directory.

**Reply by VincentVToscano** [reply]
I have to check to see if it does. Will let ya know.

**Comment by mischnic** [comment]
`parcel start` doesn't use any manually defined target by default (among other reasons, because it wouldn't know which target to use if multiple are defined).

You can add `--target frontend` to the dev command to override that.

**Reply by VincentVToscano** [reply]
OK, so `parcel start` ignores and manually defined target objects a user defines in `package.json`. Taking you r suggestion, I should add CLI flag to allow for the override, and point to my defined `frontend` target. Thank you

**Reply by sarahrobichaud** [reply]
`Parcel@2.6.0`
`node@16.13.1`

I config looks like this 
```json

{
  "targets": {
    "resources": {
      "source": "resources/resources.html",
      "distDir": "dist/resources",
      "publicUrl": "dist/resources/"
    }
  }
}
``` 
I've tried `parcel --target resources --no-hmr` without success.

> With the `--target` flag can I do multiple at once?

EDIT: Yes [Parcel docs](https://parceljs.org/features/cli/#parameters)

> parcel start doesn't use any manually defined target by default (among other reasons, because it wouldn't know which 
target to use if multiple are defined).

It takes the correct `source` and `publicPath`, just not `distDir`, is it because its the only one defined?
Also any ideas?

EDIT: `--target` flag works with `build` but not `serve` nor `watch`.

I saw this mentionned in #8105, will just try to work around it for now.

**Comment by nitipit** [comment]
Just put config in `package.json` like this in the first **target** to make your build command follow `--dist-dir`
```
"targets": {
    "": {
            "outputFormat": "global",
            "optimize": true
        }
}
```
